Classes/FrontendEnvironment/Tsfe.php

```diff
@@ -377,6 +380,15 @@ protected function getPidToUseForTsfeInitialization(int $pidToUse, ?int $rootPag
         if (isset($rootPageId)) {
             return $rootPageId;
         }
+
+        // Check for recursion that can happen if the root page is a sysfolder with a typoscript template
+        if ($pidToUse == $incomingPidToUse && $rootPageId == $incomingRootPageId) {
```

Comment:
@BrandSven 
Please fix this scrutinizer issue by using strict comparison operator.

Reply:
Should be fixed now